runtime/tutorials/how_to_with_npm/react.md

```diff
@@ -65,180 +89,205 @@ router
 
-```js
-import { defineConfig } from "npm:vite@^3.1.3";
-import react from "npm:@vitejs/plugin-react@^2.1";
+We'll use [`react-router-dom`](https://reactrouter.com/en/main) to build out
+some routing logic, so we'll need to add the `react-router-dom` dependency to
+your project. In the project root run:
 
-import "npm:react@^18.2";
-import "npm:react-dom@^18.2/client";
-import "npm:react-router-dom@^6.4"; // Add this line
-
-// https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [react()],
-});
+```shell
+deno add npm:react-router-dom
 ```
 
-Once we add the dependencies there, we can import them without `npm:` specifier
-throughout our React app.
+Update the `/src/main.tsx` file to import and use the
+[`BrowserRouter`](https://reactrouter.com/en/main/router-components/browser-router)
+component from `react-router-dom`:
 
-Next, let's go to `src/App.jsx` and add our routing logic:
-
-```jsx
+```ts title="main.tsx"
 import React from "react";
-import {
-  BrowserRouter as Router,
-  Navigate,
-  Route,
-  Routes,
-} from "react-router-dom";
-import Index from "./pages/Index.jsx";
-import Dinosaur from "./pages/Dinosaur.jsx";
-
-export default function App(props) {
-  return (
-    <Router>
-      <Routes>
-        <Route exact path="/" element={<Index />} />
-        <Route exact path="/:dinosaur" element={<Dinosaur />} />
-        <Route path="*" element={<Navigate to="/" />} />
-      </Routes>
-    </Router>
-  );
-}
+import ReactDOM from "react-dom/client";
+import App from "./App";
+import { BrowserRouter } from "react-router-dom";
+import "./index.css";
+
+ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
+  <BrowserRouter>
+    <App />
+  </BrowserRouter>,
+);
 ```
 
-Next, let's add the `<Index>` and `<Dinosaur>` pages.
+## Create the routes and types
 
-## Add pages
+We'll create two pages: `Index` and `Dinosaur`. The `Index` page will list all
+the dinosaurs and the `Dinosaur` page will show details of a specific dinosaur.
 
-There will be two pages in this app:
+Create a `pages` folder in the `src` directory and inside that create two files:
+`Index.tsx` and `Dinosaur.tsx`.
 
-- `src/pages/Index.jsx`: our index page, which lists all of the dinosaurs
-- `src/pages/Dinosaur.jsx`: our dinosaur page, which shows details of the
-  dinosaur
+### Types
 
-We'll create a `src/pages` folder and create the `.jsx` files:
+Both pages will use the `Dino` type to describe the shape of data they're
+expecting from the API, so let's create a `types.ts` file in the project root:
 
-```shell
-mkdir src/pages && touch src/pages/Index.jsx src/pages/Dinosaur.jsx
+```ts title="types.ts"
+export type Dino = { name: string; description: string };
 ```
 
-Let's start with `<Index>`. This page will `fetch` at `localhost:8000/api` and
-render that through JSX.
+### Index.tsx
+
+This page will fetch the list of dinosaurs from the API and render them as
+links:
 
-```jsx
+```ts title="Index.tsx"
```

Comment:
It's unusual to use an uppercase `Index.tsx` instead of `index.tsx` imo.

````suggestion
```tsx title="index.tsx"
````